compiler/src/dotty/tools/dotc/transform/init/Summarization.scala

```diff
@@ -213,6 +214,19 @@ object Summarization {
     else summary
   }
 
+  private def effectsOfType(tp: Type, source: Tree)(implicit env: Env): Effects =
```

Comment:
Is this function calculating the effects of constructing an instance of type `tp`?

Reply:
It's calculating field access effects in types. In Scala we can have types like `p.f.T`, here we summarize the effects for `p.f`.